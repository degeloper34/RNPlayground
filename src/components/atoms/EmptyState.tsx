import {MaterialCommunityIcons} from "@expo/vector-icons";
import {View, StyleSheet} from "react-native";
import Colors from "../../constants/Colors";
import {CustomText} from "./CustomText";

export function EmptyState({emptyStateText}: {emptyStateText: string}) {
  return (
    <View style={styles.container}>
      <MaterialCommunityIcons
        name="alert-octagram"
        size={100}
        color={Colors.mainYellow}
      />
      <CustomText
        text={emptyStateText}
        type={"medium"}
        textColor={Colors.mainNight}
        fontSize={14}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
});
