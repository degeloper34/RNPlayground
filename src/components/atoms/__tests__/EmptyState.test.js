import renderer from "react-test-renderer";
import React from "react";
import {EmptyState} from "../EmptyState";

describe("EmptyState", () => {
  it("renders correctly cart EmptyState", () => {
    const component = renderer
      .create(<EmptyState emptyStateText="There is no product in your cart." />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
