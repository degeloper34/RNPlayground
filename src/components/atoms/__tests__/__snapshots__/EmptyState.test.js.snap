// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders correctly cart EmptyState 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#21325E",
          "fontFamily": "sora-medium",
          "fontSize": 14,
          "textDecorationLine": "none",
        },
      ]
    }
  >
    There is no product in your cart.
  </Text>
</View>
`;
